src/management/index.js

```diff
@@ -144,12 +144,17 @@ class ManagementClient {
       }
 
       this.tokenProvider = new ManagementTokenProvider(config);
-    } else if (typeof options.token !== 'string' || options.token.length === 0) {
+    } else if (
+      typeof options.token !== 'function' &&
+      (typeof options.token !== 'string' || options.token.length === 0)
+    ) {
       throw new ArgumentError('Must provide a token');
     } else {
       this.tokenProvider = {
         getAccessToken() {
-          return Promise.resolve(options.token);
+          return typeof options.token !== 'string'
+            ? options.token()
+            : Promise.resolve(options.token);
         },
       };
       managerOptions.headers['Authorization'] = `Bearer ${options.token}`;
```

Comment:
This doesn't work if `options.token` is a function, but since there is slways a `tokenProvider` you can delete this line